File: prototype/rails/src/app/domain/reports/payout_summary.rb
require_relative "../money"

module Domain
  module Reports
    # What one payout run came to, for the line the portal reports back.
    PayoutSummary = Data.define(:count, :total) do
      def self.of(amounts)
        new(count: amounts.length, total: amounts.sum(Money.from_cents(0)))
      end
    end
  end
end

File: prototype/php/src/app/Domain/Reports/PayoutSummary.php
<?php

declare(strict_types=1);

namespace App\Domain\Reports;

use App\Domain\Money\Money;

final readonly class PayoutSummary
{
    private function __construct(public int $count, public Money $total) {}

    /**
     * @param  list<Money>  $amounts  one entry per payout a run wrote
     */
    public static function of(array $amounts): self
    {
        return new self(
            count($amounts),
            array_reduce($amounts, fn (Money $total, Money $amount): Money => $total->add($amount), Money::zero()),
        );
    }
}
